docs/source/upgrade-guide/index.md

```diff
@@ -53,6 +53,112 @@ or use Webpack plugins, you might need to make adjustments.
 Razzle has been upgraded to version `4.2.18`.
 It is recommended that you update your project's dependency on Razzle to this version in order to avoid duplication.
 
+### TypeScript support in Volto
+
+```{versionadded} 17.0.0-alpha.22
+We added full support of TypeScript in Volto core.
+No existing code has been migrated.
+You still have the freedom to develop using JavaScript.
+It's your choice whether you want to do it in TypeScript.
+In the past, you could add TypeScript support for Volto add-ons.
+It's also available for projects now, if you upgrade dependencies (see next section).
+```
```

Comment:
Let's simplify this block, moving most content outside of the block.

````suggestion
```{versionadded} 17.0.0-alpha.22
```

We added full support of TypeScript in Volto core.

No existing code has been migrated.
You may write code in either JavaScript or TypeScript.
The choice is yours.

Previously developers had the option to support TypeScript only in their Volto add-ons.
Now it's available in Volto projects, as long as you upgrade your project dependencies.
````